### ISSUE TITLE
ptyr files contain no data in nd-classes

### ISSUE THREAD

**Issue by alexbjorling** [ptycho/ptypy#135]
Just noticed that the `ptyr` files produced by the `nd-classes` branch (with `minimal_prep_and_run.py` after enabling `autosave`) contain no object/probe data. Looking in the debugger it's not clear to me why, the data structures in `Ptycho.save_run()` look very similar. Unfortunately I don't have time to investigate it.

**Comment by bjoernenders** [comment]
I'll have a look at it soon

**Comment by aaron-parsons** [comment]
I have also just found this. I might have a look if I get chance tomorrow.

**Comment by pierrethibault** [comment]
@bjoernenders:

@aaron-parsons and I just looked into this. We haven't found the problem but there is something fishy about lines [587-594](https://github.com/ptycho/ptypy/blob/201d756f230581e92fb72fd668a44f81db144228/ptypy/core/ptycho.py#L587-L594) in `ptycho.py` (from latest `release_candidate`): autosave can be attempted before the very first iteration, so that the default in `path.Paths.runtime` is used. Not sure if this is related with the fact that no probe/object is stored at this point...

**Comment by aaron-parsons** [comment]
Looking at this now. Have made a test which fails.

**Comment by aaron-parsons** [comment]
This is just because the h5rw doesn't support an OrderedDict to be saved, it was actually failing silently because:io.h5options['UNSUPPORTED'] = 'ignore', setting this to fail showed up the issue. 

It's a trivial fix, but I'm just knocking up a quick set of tests to make sure it doesn't happen again, and that we proof h5rw against failure in the future. Hopefully will be done today.

**Comment by aaron-parsons** [comment]
I'm fixing this on the fix_saving branch. It turned out that both ordereddict and rec array weren't supported. OrderedDict was an easy one, but rec array with unicode strings wasn't supported by h5py. I used the fix Andrew gives here: https://github.com/h5py/h5py/issues/289 for the recarray.

I've pushed this (plus tests to make sure this doesn't happen again), but I want to fix the loading before completing the PR, since it's obviously also broken there. This I will attempt to do tomorrow.
